service-base/src/main/java/org/eclipse/hono/service/HealthCheckServer.java

```diff
@@ -93,14 +93,13 @@ private void prepareHealthCheck() {
      * 
      * @param serviceInstance instance of the service which's checks should be registered.
      */
-    // TODO: change parameter to AbstractServiceBase and remove casts (see: https://github.com/eclipse/hono/pull/287).
-    public void registerHealthCheckResources(Object serviceInstance) {
+    public void registerHealthCheckResources(HealthCheckProvider serviceInstance) {
```

Comment:
is this method supposed to be overridden?
If not, I'd prefer to make the whole class final ...